### PR TITLE
Bug 1508724 - Improve logging when using an incorrect tag

### PR DESCRIPTION
**Pull request by rthallisey**
**Describe what this PR does and why we need it**:
If you set an incorrect tag for your images in the broker config, the logging
does not indicate what the error is.  Instead, make a suggestion
about the error and print out the request body.

Changes proposed in this pull request
 - Improve logging when an incorrect tag is used

### CODE REVIEW

File: pkg/registries/adapters/adapter.go [hunk 1]
```diff
@@ -91,8 +93,14 @@ func imageToSpec(log *logging.Logger, req *http.Request, image string) (*apb.Spe
...
 
-	err = json.NewDecoder(resp.Body).Decode(&hist)
+	// resp.Body is an io.Reader, which are a one time use.  Save the
+	// contents to a byte[] for debugging, then remake the io.Reader for the
+	// JSON decoder.
+	debug, _ := ioutil.ReadAll(resp.Body)
+	r := bytes.NewReader(debug)
+	err = json.NewDecoder(r).Decode(&hist)
 	if err != nil {
+		log.Errorf("Image '%s' may not exist in registry: %s", image, debug)
```
Comment (shawn-hurley) [inline pkg/registries/adapters/adapter.go]: I think using the registry name here, rather then the response body, might make more sense and then you can remove the copy of the byte array?

Comment (rthallisey) [reply pkg/registries/adapters/adapter.go]: This error check really is catching two error scenarios: A json decoding error and a request error.  I'm going to change this to two different checks by looking for a bad http status code.

I thought about that adding in the registry, but it's embedded in the image variable: ```http://<registry>/<image>/<tag>```.

**Comment by rthallisey** [comment]
The logging error came from https://bugzilla.redhat.com/show_bug.cgi?id=1508724
Do we want an addition bz for this?

**Comment by shawn-hurley** [comment]
I just re-use that BZ, I think this is something worth merging IMO

**Pull request merged**